cannon/mipsevm/page.go

```diff
@@ -1,12 +1,39 @@
 package mipsevm
 
 import (
+	"bytes"
+	"compress/zlib"
+	"encoding/base64"
 	"encoding/hex"
+	"encoding/json"
 	"fmt"
+	"io"
+	"regexp"
+	"sync"
 
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
+var zlibWriterPool = sync.Pool{
+	New: func() any {
+		var buf bytes.Buffer
+		return zlib.NewWriter(&buf)
+	},
+}
+
+func detectEncoding(data []byte) (string, error) {
+	// Regular expressions to check for base64 and hex patterns
+	base64Pattern := "^[A-Za-z0-9+/]*={0,2}$"
+	hexPattern := "^[0-9a-fA-F]*$"
+
+	if regexp.MustCompile(hexPattern).MatchString(string(data)) {
+		return "hex", nil
+	} else if regexp.MustCompile(base64Pattern).MatchString(string(data)) {
+		return "base64", nil
+	}
+	return "", fmt.Errorf("Unable to determine encoding for data: %s", data)
+}
+
 type Page [PageSize]byte
 
 func (p *Page) MarshalText() ([]byte, error) {
```

Comment:
We should be able to remove this function. It's unused now that `Page` implements `MarshalJSON`